lightning/src/ln/channelmanager.rs

```diff
@@ -3353,6 +3346,46 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		for (source, payment_hash, reason) in timed_out_htlcs.drain(..) {
 			self.fail_htlc_backwards_internal(self.channel_state.lock().unwrap(), source, &payment_hash, reason);
 		}
+	}
+
+	/// Updates channel state to take note of transactions which were confirmed in the given block
+	/// at the given height.
+	///
+	/// Note that you must still call update_best_block with the block information here.
+	///
+	/// XXX: Note that in the case that a transaction which was provided via this method is
+	/// reorganized out of the best chain, the only current way to correctly handle the any
+	/// associated channel force-closure is to walk the chain back from the current tip with
+	/// repeated `block_disconnected` calls, followed by an `update_best_block` call.
+	pub fn transactions_confirmed(&self, header: &BlockHeader, height: u32, txdata: &TransactionData) {
+		// Note that we MUST NOT end up calling methods on self.chain_monitor here - we're called
+		// during initialization prior to the chain_monitor being fully configured in some cases.
+		// See the docs for `ChannelManagerReadArgs` for more.
+
+		let block_hash = header.block_hash();
+		log_trace!(self.logger, "{} transactions included in block {} at height {} provided", txdata.len(), block_hash, height);
+
+		let _persistence_guard = PersistenceNotifierGuard::new(&self.total_consistency_lock, &self.persistence_notifier);
+		self.do_block_event(height, |channel| channel.transactions_confirmed(&block_hash, height, txdata, &self.logger).map(|a| (a, Vec::new())));
+	}
+
+	/// Updates channel state with the current best blockchain tip. You should attempt to call this
+	/// quickly after a new block becomes available, however it does not need to be called for each
+	/// new block.
```

Comment:
"It does not need to be called for each new block" What about `check_get_funding_locked()` to account for confirmations ?

Reply:
I'm not really sure what you're asking? `check_get_funding_locked` handles skipped blocks fine, or, it should?

Reply:
Better now. I think you can add a small comment on `check_get_funding_locked` "Confirmations are accounted with an absolute height and not incrementally. This method might be only be called when new chain tip reach, after multiple block connections".